webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/toolbar/ActivityComposerDrawer.vue

```diff
@@ -444,6 +458,90 @@ export default {
     },
+    resetRichEditorData() {
+      let message = this.ckEditorInstance.getMessage();
+      if (this.audience) {
+        if (message?.length) {
+          const mentionedUsers = message.match(/@([A-Za-z0-9_'.+-]+)/g).map(a => a.replace('@', ''));
+          if (mentionedUsers?.length) {
+            // Clear suggester cache
+            document.dispatchEvent(new CustomEvent('suggester-clear-people-cache'));
+            Promise
+              .all(mentionedUsers.map(username => {
+                return this.$identityService.getIdentityByProviderIdAndRemoteId('organization', username)
+                  .then(identity => identity?.profile && this.$spaceService.isSpaceMember(this.audience.spaceId, username)
+                    .then(data => {
+                      if (data) {
+                        const profile = identity.profile;
+                        profile.isMember = data?.isMember === 'true';
+                        return profile;
+                      } else {
+                        return null;
+                      }
+                    }));
+              }))
+              .then(userProfiles => userProfiles.filter(p => p))
+              .then(userProfiles => {
+                userProfiles.forEach(profile => {
+                  if (profile.isMember) {
+                    message = message.replace(new RegExp(`@${profile.username}`, 'g'), ExtendedDomPurify.purify(`
+                      <span class="atwho-inserted" data-atwho-at-query="@${profile.username}" data-atwho-at-value="${profile.username}" contenteditable="false">
+                        <span class="exo-mention">${profile.fullname}${profile.isExternal === 'true' ? ` (${  this.$t('UsersManagement.type.external')  })` : ''}<a href="#" class="remove"><i class="uiIconClose uiIconLightGray"></i></a></span>
+                      </span>
+                    `));
+                  } else {
+                    message = message.replace(new RegExp(`@${profile.username}`, 'g'), ExtendedDomPurify.purify(`
+                      <span class="atwho-inserted" data-atwho-at-query="@${profile.username}" data-atwho-at-value="" contenteditable="false" title="${this.$t('activity.composer.invalidUser.message')}">
+                        <span class="exo-mention">
+                          <i aria-hidden="true" class="v-icon notranslate fa fa-exclamation-triangle theme--light orange--text error-color" style="font-size: 16px;">
+                          </i>
+                          <del>${profile.fullname}${profile.isExternal === 'true' ? ` (${  this.$t('UsersManagement.type.external')  })` : ''}</del>
+                          <a href="#" class="remove">
+                            <i class="uiIconClose uiIconLightGray"></i>
+                          </a>
+                        </span>
+                      </span>
+                    `));
+                    this.containInvalidUsers = true;
+                  }
+                });
```

Comment:
It would be better to set a new value of `containInvalidUsers` when switching audiences, else, the warning can be displayed whereas it could display a wrong information to end user.

```suggestion
                  }
                });
                this.containInvalidUsers = userProfiles.find(p => !p.isMember);
```

Reply:
I have set a false value for the variable when switching audience

Reply:
ok, there are two `if` in fact, we will have to set a value when switch the audience to "my network" by example.

Reply:
BTW, it doesn't contain all time invalid users since we can switch to a space where the mentioned users are member as well, thus the message shouldn't be displayed.

Reply:
done!